todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskFragment.kt

```diff
@@ -55,21 +54,22 @@ class AddEditTaskFragment : Fragment() {
         viewDataBinding = AddtaskFragBinding.bind(root).apply {
             viewmodel = (activity as AddEditTaskActivity).obtainViewModel()
         }
+        viewDataBinding.setLifecycleOwner(this.viewLifecycleOwner)
         setHasOptionsMenu(true)
         retainInstance = false
         return viewDataBinding.root
     }
 
     private fun setupFab() {
-        activity.findViewById<FloatingActionButton>(R.id.fab_edit_task_done).apply {
-            setImageResource(R.drawable.ic_done)
-            setOnClickListener { viewDataBinding.viewmodel?.saveTask() }
+        activity?.findViewById<FloatingActionButton>(R.id.fab_edit_task_done)?.let {
```

Comment:
Why? apply is right here (or run)

Reply:
I prefer not to change receivers, especially in samples. It makes things harder to read.